spark/README.md

```diff
@@ -0,0 +1,151 @@
+## Spice on Apache Spark
+
+Spice can read data straight from a Spark instance. This guide will create an app, configure Spark to run locally, load and query a dataset. It assumes:
+
+- Spice is installed (see the [Getting Started](https://docs.spiceai.org/getting-started) documentation).
+- Spark Connect Server is running locally (refer to the [Quickstart: Spark Connect](https://spark.apache.org/docs/latest/api/python/getting_started/quickstart_connect.html) to launch spark server with spark connect)
+- pyspark is installed in your python environment. (see [Installation](https://spark.apache.org/docs/latest/api/python/getting_started/install.html) for downloading pyspark in your python environment)
+
+1. Initialise a Spice app
+
+   ```shell
+   spice init spark_demo
+   cd spark_demo
+   ```
+
+2. Start the Spice runtime
+
+   ```shell
+   >>> spice run
+   Spice.ai runtime starting...
+   2024-05-20T23:54:42.323695Z  INFO spiced: Metrics listening on 127.0.0.1:9000
+   2024-05-20T23:54:42.325278Z  INFO runtime::opentelemetry: Spice Runtime OpenTelemetry listening on 127.0.0.1:50052
+   2024-05-20T23:54:42.327243Z  INFO runtime::http: Spice Runtime HTTP listening on 127.0.0.1:3000
+   2024-05-20T23:54:42.327255Z  INFO runtime::flight: Spice Runtime Flight listening on 127.0.0.1:50051
+   ```
+
+3. Create the Sample Dataset in Spark
+
+This Quickstarts use NYC taxi trip parquet data from [TLC Trip Record Data](https://www.nyc.gov/site/tlc/about/tlc-trip-record-data.page) to create a sample table in Spark.
+
+Download the NYC taxi trip parquet file using the following command
+
+```shell
+wget https://d37ci6vzurychx.cloudfront.net/trip-data/yellow_tripdata_2024-01.parquet
+```
+
+Change the `parquet_file_path` in python script to the absolute file path where `yellow_tripdata_2024-01.parquet` is located. Run the following python script in the python environment that already have `pyspark` downloaded.
+
+```python
```

Comment:
Please add instructions to setup virtual env and install all dependencies
```bash
python3 -m venv .venv
source .venv/bin/activate
pip install pyspark pandas ...
```
https://spark.apache.org/docs/latest/api/python/getting_started/install.html#dependencies

Reply:
Dependencies updated in pre-requisites. The python virtual environment creation / usage should be left for user choice, since both `python3 -m venv .venv` and `conda` can be used for creating virtual environment. Here we can assume that the user have knowledge about using python virtual environment